app/modules/platform/saga/api/apiPostSigninToTokenAndGetUserAuth.js

```diff
@@ -24,23 +25,23 @@ const apiPostSigninToTokenAndGetUserAuth = function* (
     const responseData: ApiResponseData = yield call(api.token.postSignin, email, password);
 
     // Get flow to stop complaining about token possibly being NULL
-    if (responseData.token == null) {
-      throw new Error(`This shouldn't happen`);
-    }
+    if (responseData.token == null) throw new Error(`This shouldn't happen`);
+
+    if (responseData.body == null) throw new UnexpectedEmptyResponseError(`Unexpected empty response data`);
```

Comment:
- I think it would be more practical to have a single type of error for the token or the body being missing (`InvalidHttpResponseError` or something like that), since these two are basically the same kind of problem.
- If the error message 'Unexpected response data' is the same every time this error is thrown, it should be set as the default message in the error's constructor, so that the developer doesn't have to type it every time. See `NotYetImplementedError` for reference.
- If you throw a new error somewhere in the code, don't forget to add a test for it.

Reply:
They might be the same problem (e.g. something is missing in the response), but they stem from different causes. A missing token might indicate a problem with the authentication, but a missing body might indicate something different entirely.

Reply:
Should I add a `MissingTokenError`?

Reply:
Correct me if I'm wrong, but I don't think it matters for the frontend what exactly went wrong in the backend; it only knowns that it did not get back the response it expected. Should the frontend react differently to a missing token and to a missing body?

Reply:
It's actually not that important to have this distinction; once the single-token authentication system gets rewritten into a dual-token mechanism, the authorization response header (token) will disappear anyway.

Reply:
Then I think for now using a single type of error for both is the most practical way to go.